src/main/java/io/github/nucleuspowered/nucleus/modules/warn/commands/WarnCommand.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * This file is part of Nucleus, licensed under the MIT License (MIT). See the LICENSE.txt file
+ * at the root of this project for more details.
+ */
+package io.github.nucleuspowered.nucleus.modules.warn.commands;
+
+import com.google.inject.Inject;
+import io.github.nucleuspowered.nucleus.ChatUtil;
+import io.github.nucleuspowered.nucleus.Util;
+import io.github.nucleuspowered.nucleus.api.data.WarnData;
+import io.github.nucleuspowered.nucleus.argumentparsers.TimespanArgument;
+import io.github.nucleuspowered.nucleus.internal.PermissionRegistry;
+import io.github.nucleuspowered.nucleus.internal.annotations.*;
+import io.github.nucleuspowered.nucleus.internal.command.CommandBase;
+import io.github.nucleuspowered.nucleus.internal.permissions.PermissionInformation;
+import io.github.nucleuspowered.nucleus.internal.permissions.SuggestedLevel;
+import io.github.nucleuspowered.nucleus.modules.warn.config.WarnConfigAdapter;
+import io.github.nucleuspowered.nucleus.modules.warn.handlers.WarnHandler;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.api.command.CommandSource;
+import org.spongepowered.api.command.args.CommandContext;
+import org.spongepowered.api.command.args.CommandElement;
+import org.spongepowered.api.command.args.GenericArguments;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.entity.living.player.User;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.channel.MessageChannel;
+import org.spongepowered.api.text.channel.MutableMessageChannel;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+@Permissions(suggestedLevel = SuggestedLevel.MOD)
+@NoWarmup
+@NoCooldown
+@NoCost
+@RegisterCommand({"warn", "warning", "addwarning"})
+public class WarnCommand extends CommandBase<CommandSource> {
+
+    public static final String notifyPermission = PermissionRegistry.PERMISSIONS_PREFIX + "warn.notify";
+    private final String bypassMaximumLength = PermissionRegistry.PERMISSIONS_PREFIX + "warn.bypass";
+    private final String warningExempt = PermissionRegistry.PERMISSIONS_PREFIX + "warn.exempt";
```

Comment:
The private static variables should be simply obtained using `permissions.testSuffix(...)` so if we change the permission base later, everything changes as expected. The public one, I need to think of a better way to do it - but that's my problem.